tests/ui/async-await/future-sizes/async-awaiting-fut.rs

```diff
@@ -1,4 +1,4 @@
-// compile-flags: -Z print-type-sizes --crate-type lib
+// compile-flags: -Z print-type-sizes --crate-type lib -Z mir-upvar-to-local-prop=true -Z mir-inline-future-into-future=true
```

Comment:
Could you use two revisions here to test the default and with the flags?

Reply:
done in https://github.com/rust-lang/rust/pull/108590/commits/48758c465f94f261232bda8a3ccb128049406564 , https://github.com/rust-lang/rust/pull/108590/commits/183b341e74e164328550dd1d04b03563233a7274 , and https://github.com/rust-lang/rust/pull/108590/commits/1e33269376dfcc7fb3f5a041d7bb166ca0209870